doc/kuksa_use_cases.md

```diff
@@ -0,0 +1,214 @@
+
+# Use cases:
+
+## 1. Kuksa client subscribes on changed sensor values or actuator current values of all signals
+
+**Primary Actor:** Kuksa client
+
+**Secondary Actors:** Databroker, Provider, CAN 
+
+**Priority**: High
+
+**Preconditions:**
+ * Provider open CAN Bus connection and start receiving signal values.
+ * Provider sets sensor values or actuator current values to Databroker atomically. -> NFR (1.7)
+ * Kuksa client contains a valid authentication token to connect and perform calls to databroker.
+
+**Special requirements:**
+ * The use case must meet the Performance NRF (3.17)
+ * What if provider died?
+
+**Assumptions:**
+ * Client wants to subscribe to all the possible sensor values or actuator current values to test special requirements.
+ * Provider set new signals values to Databroker as fast as receives data from CAN bus.
+
+**Postconditions:**
+ * A subscription is created, and client receives streaming of signals.
+
+**Sequence diagram:**
+
+![Client Subscribes](./diagrams/Client_subscribes.drawio.png)
+
+**Basic Flow:**
+1.	Kuksa client subscribe to all the sensor values or current values of actuators.
+2.	Kuksa client start receiving streaming of data.
+3.	Use case finishes.
+
+**Alternative Flows**:
+* Flow 1.
+2.	Databroker return error response since provider is not connected???
+3.	Use case finishes.
+
+**Exceptions:**
+
+
+## 2. Kuksa client sets a target value for an actuator.
+
+**Primary Actor:** Kuksa client
+
+**Secondary Actors:** Databroker, Provider, CAN Bus
+
+**Priority:** High
+
+**Preconditions:**
+* Provider is connected to some CAN Bus.
+* Provider started sending signals current values to Databroker before the use case starts.
+* Provider subscribes to target values coming from Databroker.
+* Kuksa client contains a valid authentication token to connect and perform calls to databroker.
+
+**Special requirements:**
+
+**Assumptions:**
+
+**Postconditions:**
+* Client receives a response whose status indicates the operation was successfully executed.
+* Databroker stores on his database the new target value for the actuator.
+* Provider writes the new target value in CAN bus.
+
+**Sequence diagram:**
+![Client Actuate](./diagrams/Client_actuate.png)
+
+**Basic Flow:**
+1.	Kuksa client calls actuate with a valid actuator path and keeps waiting for a response.
+2.	Databroker checks if a Provider is connected.
+3.	Databroker changes its internal target value for the corresponding actuator path with the current timestamp. (Assumption -> Current value remains the same)
+4.	Databoker notifies all subscriptions to target values.
```

Comment:
Question: Do we allow multiple subscribers to the target value of which only one is registered as the provider?